workspaces/spectacle/src/index.ts

```diff
@@ -207,6 +209,7 @@ export async function makeSpectacle(opticContext: IOpticContext) {
     shapeViewerProjection: any,
     contributionsProjection: ContributionsProjection;
 
+  // TODO: consider debouncing reloads (head and tail?)
```

Comment:
Chokidar already comes with a 2 second debounce to deal with fast sequential operations like Git rebases, atomic writes from editors, etc. That should be enough as a starting point.